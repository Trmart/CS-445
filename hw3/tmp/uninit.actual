.   .   .   .   Child: 0  Const 4 [line: 8]
.   .   .   .   Child: 0  Id: g [line: 7]
.   .   .   Child: 0  Id: i [line: 7]
.   .   .   Child: 0  Id: i [line: 8]
.   .   .   Child: 0  Id: z [line: 5]
.   .   .   Child: 0  Id: z [line: 6]
.   Child: 0  Parm: x of type int [line: 1]
.   .   Child: 0  Var: z of type int [line: 3]
.   .   Child: 1  Assign: <= [line: 5]
.   Child: 1  Compound [line: 2]
.   .   .   .   Child: 1  Const 3 [line: 7]
.   .   .   .   Child: 1  Id: h [line: 8]
.   .   .   Child: 1  Id: i [line: 5]
.   .   .   Child: 1  Id: i [line: 6]
.   .   .   Child: 1  Op: + [line: 7]
.   .   .   Child: 1  Op: + [line: 8]
Func: ( returns type void [line: 1]
.   .   Sibling: 1  Assign: <= [line: 6]
.   .   Sibling: 1  Var: g of type int [line: 3]
.   .   Sibling: 2  Assign: <= [line: 7]
.   .   Sibling: 2  Var: h of type int [line: 3]
.   .   Sibling: 3  Assign: <= [line: 8]
.   .   Sibling: 3  Var: i of type int [line: 3]
.   .   Sibling: 4  Var: j of type int [line: 3]
.   .   Sibling: 5  Var: k of type int [line: 3]
.   .   Sibling: 6  Var: l of type int [line: 3]
.   .   Sibling: 7  Var: m of type int [line: 3]
.   .   Sibling: 8  Var: z of type int [line: 3]
.   .   Sibling: 9  Var: z of type int [line: 3]
