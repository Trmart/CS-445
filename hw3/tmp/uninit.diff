1,15c1,29
< ERROR(13): Symbol 'dog' is not declared.
< ERROR(14): Cannot use function 'main' as a variable.
< ERROR(3): Symbol 'z' is already declared at line 3.
< ERROR(3): Symbol 'z' is already declared at line 3.
< ERROR(LINKER): A function named 'main()' must be defined.
< Number of errors: 5
< Number of warnings: 8
< WARNING(10): Variable 'k' may be uninitialized when used here.
< WARNING(11): Variable 'l' may be uninitialized when used here.
< WARNING(12): Variable 'm' may be uninitialized when used here.
< WARNING(1): The variable 'x' seems not to be used.
< WARNING(5): Variable 'i' may be uninitialized when used here.
< WARNING(7): Variable 'g' may be uninitialized when used here.
< WARNING(8): Variable 'h' may be uninitialized when used here.
< WARNING(9): Variable 'j' may be uninitialized when used here.
---
> .   .   .   .   Child: 0  Const 4 [line: 8]
> .   .   .   .   Child: 0  Id: g [line: 7]
> .   .   .   Child: 0  Id: i [line: 7]
> .   .   .   Child: 0  Id: i [line: 8]
> .   .   .   Child: 0  Id: z [line: 5]
> .   .   .   Child: 0  Id: z [line: 6]
> .   Child: 0  Parm: x of type int [line: 1]
> .   .   Child: 0  Var: z of type int [line: 3]
> .   .   Child: 1  Assign: <= [line: 5]
> .   Child: 1  Compound [line: 2]
> .   .   .   .   Child: 1  Const 3 [line: 7]
> .   .   .   .   Child: 1  Id: h [line: 8]
> .   .   .   Child: 1  Id: i [line: 5]
> .   .   .   Child: 1  Id: i [line: 6]
> .   .   .   Child: 1  Op: + [line: 7]
> .   .   .   Child: 1  Op: + [line: 8]
> Func: ( returns type void [line: 1]
> .   .   Sibling: 1  Assign: <= [line: 6]
> .   .   Sibling: 1  Var: g of type int [line: 3]
> .   .   Sibling: 2  Assign: <= [line: 7]
> .   .   Sibling: 2  Var: h of type int [line: 3]
> .   .   Sibling: 3  Assign: <= [line: 8]
> .   .   Sibling: 3  Var: i of type int [line: 3]
> .   .   Sibling: 4  Var: j of type int [line: 3]
> .   .   Sibling: 5  Var: k of type int [line: 3]
> .   .   Sibling: 6  Var: l of type int [line: 3]
> .   .   Sibling: 7  Var: m of type int [line: 3]
> .   .   Sibling: 8  Var: z of type int [line: 3]
> .   .   Sibling: 9  Var: z of type int [line: 3]
