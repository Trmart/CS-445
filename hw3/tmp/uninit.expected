ERROR(13): Symbol 'dog' is not declared.
ERROR(14): Cannot use function 'main' as a variable.
ERROR(3): Symbol 'z' is already declared at line 3.
ERROR(3): Symbol 'z' is already declared at line 3.
ERROR(LINKER): A function named 'main()' must be defined.
Number of errors: 5
Number of warnings: 8
WARNING(10): Variable 'k' may be uninitialized when used here.
WARNING(11): Variable 'l' may be uninitialized when used here.
WARNING(12): Variable 'm' may be uninitialized when used here.
WARNING(1): The variable 'x' seems not to be used.
WARNING(5): Variable 'i' may be uninitialized when used here.
WARNING(7): Variable 'g' may be uninitialized when used here.
WARNING(8): Variable 'h' may be uninitialized when used here.
WARNING(9): Variable 'j' may be uninitialized when used here.
